### DISCUSSION TITLE
Does django-unicorn support file uploads

### DISCUSSION THREAD

**Discussion by nyashaChiza** [django-commons/django-unicorn#729] [Q&A]
### Question: Does Django-Unicorn Support File Uploads?

I'm currently working on a Django project where I'm using Django-Unicorn for a reactive front-end. I need to implement file uploads as part of my component's functionality, specifically to handle image uploads for steganography encoding. 

I've set up a form using the standard `<input type="file" unicorn:model="cover_image" />`, but I'm running into issues where the file is being passed as a string (with a fake path like `C:\fakepath\filename.png`) instead of an `InMemoryUploadedFile` object or similar file-like object that I can work with on the server side.

Here's a brief overview of my setup:

**HTML Template:**
```html
{% extends 'layouts/base.html' %}

{% block content %}
<div unicorn:view="EncodeComponentView">
    <div class="form-group">
        <label for="message">Message to hide:</label>
        <input unicorn:model="message" class="form-control" type="text" id="message" /><br />
    </div>

    <div class="form-group">
        <label for="cover_image">Select a cover image:</label>
        <input type="file" id="cover_image" unicorn:model="cover_image" class="form-control"/>
    </div>

    <button unicorn:click="encode" class="btn btn-primary form-control">Encode</button>
</div>
{% endblock %}
```

**Unicorn Component:**
```python
from django_unicorn.components import UnicornView
from stegano import lsb
from django.core.files.base import ContentFile
from PIL import Image
import io

class EncodeComponentView(UnicornView):
    stego_image_url = None
    message = ""
    cover_image = None  # Expected to be an InMemoryUploadedFile or similar

    def encode(self):
        if self.cover_image and self.message:
            image_file = self.cover_image.read()  # This line fails because cover_image is a string

            # Further processing...
```

**Issue:**
When I submit the form, `cover_image` is passed as a string path rather than an actual file object. This string is often something like `"C:\fakepath\filename.png"`, which is not the real file path and is unusable in the back-end. I need to know how to properly handle file uploads in Django-Unicorn to get an `InMemoryUploadedFile` or similar object that I can work with.

**Questions:**
1. Does Django-Unicorn support file uploads directly? 
2. If so, what is the correct way to bind a file input to a component attribute to ensure the file is properly uploaded and accessible as an `InMemoryUploadedFile` or similar in the Unicorn view?
3. Are there any additional configurations or steps required to make file uploads work with Django-Unicorn?

Any guidance or examples would be greatly appreciated.

**Comment by adamghill** [comment]
Hello @nyashaChiza, thanks for the question! File upload support has never been added to `django-unicorn`. I made a [draft PR](https://github.com/adamghill/django-unicorn/pull/167) a long time ago, but never ended up finishing it. If you would like to work on it, this PR might give an ok starting place.

**Reply by nyashaChiza** [reply]
Thanks @adamghill, let me check it out l might be able to contribute
